### PR TITLE
Unstable

### PR DESCRIPTION
**Pull request by MeidoNoHitsuji**
- generates an appropriate name for the autocomplete event
- add argument "value" of the command in the autocomplete event
- don't be afraid to cancel if you find any error :D

### CODE REVIEW

File: interactions/api/gateway.py [hunk 1]
```diff
@@ -297,17 +297,26 @@ def handle_dispatch(self, event: str, data: dict) -> None:
...
-                        if hasattr(context.data, "options"):
+                        if "options" in context.data:
                             if context.data.options:
                                 for option in context.data.options:
-                                    if option["type"] in (
-                                        OptionType.SUB_COMMAND,
-                                        OptionType.SUB_COMMAND_GROUP,
-                                    ):
-                                        if option.get("options"):
+                                    if option["type"] == OptionType.SUB_COMMAND_GROUP:
```
Comment (i0bs) [inline interactions/api/gateway.py]: I think we could build a submethod to make this check easier inside of the `handle_dispatch`:

```py
def check_sub_auto(option):
    if "options" in option:
        if option["type"] == OptionType.SUB_COMMAND:
            if sub_option.get("focused") for sub_option in option["options"]:
                _args.append(option["name"])
        elif option["type"] == OptionType.SUB_COMMAND_GROUP:
            if "options" in group_option for group_option in option["options"]:
                if sub_option.get("focused") for sub_option in group_option["options"]:
                    _name += sub_option["name"]
                    _args.append(sub_option["value"])
    elif option.get("focused"):
        _name += option{"name"]
        _args.append(option["value"])
```

This check will also respect the issue you have with changed line 307, where we suddenly ignore the actual group option. Black should also format all of this to respect the character line limit, so it theoretically will look better once pre-committed.

Comment (i0bs) [inline interactions/api/gateway.py]:
```suggestion
                                    check_sub_auto(option)
```

**Review by i0bs — CHANGES_REQUESTED** [review]
Changes to respect subcommand checking in autocompletion.

**Comment by i0bs** [comment]
Does this work with the new suggestions @MeidoNoHitsuji ?
If possible, please provide debug log.

**Comment by MeidoNoHitsuji** [comment]
Yes, this work with the new suggestions
I don't know what kind of logs ur needs)
I also added the transfer of the selection of command parameters (subcommands, groups and options) to the callback..

**Review by i0bs — APPROVED** [review]
LGTM.

**Comment by i0bs** [comment]
I'll be merging this pull request. Thank you for your contribution! Welcome to the contributors table. 😃

**Pull request merged**